File: claude/node/fiassed/models/Attachment.js
'use strict';

const { DataTypes } = require('sequelize');
const { v4: uuidv4 } = require('uuid');

module.exports = (sequelize) => {
  const Attachment = sequelize.define('Attachment', {
    id: {
      type: DataTypes.UUID,
      defaultValue: () => uuidv4(),
      primaryKey: true,
      allowNull: false
    },
    noteId: {
      type: DataTypes.UUID,
      allowNull: false,
      references: {
        model: 'notes',
        key: 'id'
      }
    },
    // Original name for display only - never used in file system paths
    originalFilename: {
      type: DataTypes.STRING(255),
      allowNull: false
    },
    // UUID-based name used for actual storage - prevents path traversal
    storedFilename: {
      type: DataTypes.STRING(255),
      allowNull: false,
      unique: true
    },
    mimeType: {
      type: DataTypes.STRING(100),
      allowNull: false
    },
    size: {
      type: DataTypes.INTEGER,
      allowNull: false,
      validate: {
        min: 1
      }
    }
  }, {
    tableName: 'attachments',
    timestamps: true,
    updatedAt: false
  });

  Attachment.associate = (models) => {
    Attachment.belongsTo(models.Note, { foreignKey: 'noteId', as: 'note' });
  };

  return Attachment;
};

File: copilot/aspnet/securable/LooseNotes/Models/Attachment.cs
using System.ComponentModel.DataAnnotations;

namespace LooseNotes.Models;

public class Attachment
{
    public int Id { get; set; }

    public int NoteId { get; set; }
    public Note Note { get; set; } = null!;

    /// <summary>Unique server-generated name (Guid + extension) — prevents path traversal.</summary>
    [Required]
    [MaxLength(500)]
    public string StoredFileName { get; set; } = string.Empty;

    /// <summary>Original file name preserved for display purposes only, never used for I/O.</summary>
    [Required]
    [MaxLength(255)]
    public string OriginalFileName { get; set; } = string.Empty;

    [Required]
    [MaxLength(100)]
    public string ContentType { get; set; } = string.Empty;

    public long FileSizeBytes { get; set; }

    public DateTimeOffset UploadedAt { get; set; } = DateTimeOffset.UtcNow;

    [Required]
    public string UploadedById { get; set; } = string.Empty;
}
